### DISCUSSION TITLE
Override repository env var

### DISCUSSION THREAD

**Discussion by jacobtomlinson** [community/community#25649] [Actions]
<p>I want to run an action on another repository. I have made a personal access token and overidden the repo name env var, however the repo name doesn’t seem to override correctly.</p>
<p>Heres a simple example of this:</p>
<p><img src="https://user-images.githubusercontent.com/1610850/82065031-75f5fb00-96c5-11ea-849e-281b81a2e368.png" alt=""></p>
<p><a href="https://github.com/dask/.github/pull/4" rel="nofollow noopener">https://github.com/dask/.github/pull/4</a></p>
<p>Is there a way to override this env var?</p>

**Comment by BrightRan** [comment]
<p><a class="mention" href="/jacobtomlinson">@jacobtomlinson</a> ,</p>
<p>Few points may be helpful to you to understand:</p>
<ol>
<li>
<p>We generally use the " <strong>env</strong>" key or the <a href="https://help.github.com/en/actions/reference/workflow-commands-for-github-actions#setting-an-environment-variable" rel="nofollow noopener"><strong>workflow command</strong></a> " <strong>::set-env name={name}::{value}</strong>" to set/update the custome environment variables. The custom environment variables set via these two ways will be added into the <a href="https://help.github.com/en/actions/reference/context-and-expression-syntax-for-github-actions#env-context" rel="nofollow noopener"><strong>env context</strong></a>.</p>
</li>
<li>
<p>The " <strong>GITHUB_REPOSITORY</strong>" is a default environment variable set by GitHub. When you try using the " <strong>env</strong>" key or the " <strong>set-env</strong>" command to change the value of a default environment variable, GitHub will prevent the value from being changed on the system, but this environment variable will be added into the  <strong>env context</strong> with the new value.</p>
</li>
<li>
<p>According to above two points, in the workflow, if you want to access the new value of an environment variable which is set/updated via the " <strong>env</strong>" key or the " <strong>set-env</strong>" command, the general method is using the expression syntax " <strong>env.&lt;env name&gt;</strong>" to access the environment variable in the  <strong>env context</strong>. For the updated default environment variable, this is the only method to access its new value, you can’t use the expression syntax that access the system environment variable to get its new value.</p>
</li>
<li>
<p>Currently, GitHub does not support a repository to run a workflow located in another repository.</p>
</li>
</ol>
<p>For example, two repositories, <strong>Repo-A</strong> and  <strong>Repo-B</strong> , and there is a workflow in  <strong>Repo-A</strong> , this workflow can’t run in  <strong>Repo-B</strong>. If you want some steps in the workflow can run based on the source code in  <strong>Repo-B</strong> , you can add a step in the workflow to <a href="https://github.com/actions/checkout" rel="nofollow noopener"><strong>checkout</strong></a> the source code from  <strong>Repo-B</strong>  into the specified directory under the workspace of the workflow, then execute the related steps the directory of  <strong>Repo-B</strong>.</p>

**Comment by jacobtomlinson** [comment]
<p><span class="mention">@brightranthanks</span> for the detailed response. Perhaps if I share more about my use case it will help.</p>
<p>When a workflow runs in RepoA I want one of the steps to act on RepoB (and RepoC, RepoD, etc). Looking at the action it uses the <strong>GITHUB_REPOSITORY</strong> env var in the API calls it makes, so I need to be able to override this. I was hoping to use a matrix to store the list of repos to act on.</p>
<p>From a permissions point of view I’ve created a personal access token which has permission to modify all repos that I need, and I can provide that in the config as a secret.</p>

**Comment by BrightRan** [comment, marked as answer]
<p><a class="mention" href="/jacobtomlinson">@jacobtomlinson</a> ,</p>
<p>Whether that action you are using in your workflow has an <strong>input parameter</strong> that allows users to specify which repository the API is executed for? Please check it.</p>
<p>If the environment variable " <strong>GITHUB_REPOSITORY</strong>" has been set as the default input and the fixed input in the action, you indeed can’t specify other repositories.</p>
<p>As a workaround, maybe you can try to copy/fork the repository of the action into your own organization. Then add an <strong>input parameter</strong> to allow users to specify which repository the API is executed for, and also replace " <strong>GITHUB_REPOSITORY</strong>" with this  <strong>input parameter</strong> in the related scripts executed in the action. Then you can use this custom action in your workflow.</p>
<p>About creating actions, you can reference the docs: <a href="https://help.github.com/en/actions/building-actions" rel="nofollow noopener">https://help.github.com/en/actions/building-actions</a></p>

**Comment by jacobtomlinson** [comment]
<p>Thanks for the response. It’s a shame it’s not possible to override this, I guess forking is my only option as you suggest.</p>

**Comment by m4heshd** [comment]
<p>Facing the same situation. It would’ve been great to be able to override a default environment variable. Fork it is then.</p>
